packages/upward-js/lib/createUpwardServer.js

```diff
@@ -49,6 +51,16 @@ async function createUpwardServer({
     const app = express();
     before(app);
     const upward = await middleware(resolve(upwardPath), env);
+
+    if (env.ENABLE_EXPRESS_SERVER_COMPRESSION === 'true') {
+        app.use(
+            compression({
+                threshold: 0
+            })
+        );
+        app.use(gzipValidationMiddleware(env));
+    }
```

Comment:
Having it in the `PWADevServer` will only run this as part of the Dev mode. If someone wants to have compression in other modes/environments as well, that wouldn't be possible with it being in `PWADevServer`. That's the reason why I have left this in the `serve` module.

Reply:
Let me rephrase; we cannot add API to `upward-js` without also updating the `upward-spec` and `upward-php` implementations. **If** I had confidence UPWARD was going to remain in the stack, this would be an option, but it would likely be better to use `upward.yml` config as opposed to an environment variable.

Given this, please move the middleware to `PWADevServer` and `utilities/serve` in buildpack. Sorry I didn't specify to also put it in the `serve` utility as well, was just pointing out where to find an example of the IOC pattern.